src/client/apps/edit/components/header/yoast/index.tsx

```diff
@@ -0,0 +1,249 @@
+  }
+
+  getBodyText = () => {
+    const {
+      article: { lead_paragraph, sections },
+    } = this.props
+    const fullText: string[] = []
+    if (lead_paragraph && lead_paragraph.length) {
+      fullText.push(lead_paragraph)
+    }
+
+    sections &&
+      sections.map(section => {
+        if (section.type === "text" && section.body) {
+          fullText.push(section.body)
+        }
+      })
+
+    return fullText.join()
+  }
+
+  toggleDrawer = () => {
+    this.setState({ isOpen: !this.state.isOpen })
+  }
+
+  onKeywordChange = e => {
+    this.props.setYoastKeywordAction(e.target.value)
+    this.resetSnippet()
+  }
+
+  resetSnippet = () => {
+    this.setSnippetFields()
+    this.snippetPreview.changedInput()
+  }
+
+  keywordIsBlank = () => {
+    const { yoastKeyword } = this.props
+    if (!yoastKeyword || (yoastKeyword && yoastKeyword.trim().length < 1)) {
+      return true
+    } else {
+      return false
+    }
+  }
+
+  generateResolveMessage = () => {
+    const issueCount: number = document.querySelectorAll("#yoast-output .bad")
+      .length
+
+    if (this.keywordIsBlank()) {
+      return " Set Target Keyword"
+    } else if (issueCount && issueCount > 0) {
+      return `${issueCount} Unresolved Issue${issueCount > 1 ? "s" : ""}`
+    } else {
+      return " Resolved"
+    }
+  }
+
+  render() {
+    const { isOpen } = this.state
+    const { yoastKeyword } = this.props
+    return (
+      <Box>
+        <YoastContainer onClick={this.toggleDrawer}>
+          Seo Analysis —
+          <ResolveMessage
+            color={
+              this.generateResolveMessage() !== " Resolved"
+                ? color("red100")
+                : color("green100")
+            }
+          >
+            {this.generateResolveMessage()}
+          </ResolveMessage>
+          <CloseIcon
+            rotation={isOpen ? 45 : 0}
+            name="follow-circle"
+            width="10px"
+            height="10px"
+            color="black"
+            title="plus-small"
+          />
+        </YoastContainer>
+        <StaticCollapse open={isOpen}>
+          <Drawer>
+            <YoastInput width={[1, 1 / 3]}>
+              <Input
+                onChange={this.onKeywordChange}
+                id="focus-keyword"
+                title="Target Keyword"
+                placeholder="A searchable term for this content"
+                value={yoastKeyword}
+              />
+            </YoastInput>
+            <YoastOutput hidden={this.keywordIsBlank()} width={[1, 2 / 3]}>
+              <YoastSnippet hidden id="yoast-snippet" />
+              <div id="yoast-output" />
+            </YoastOutput>
+          </Drawer>
+        </StaticCollapse>
+      </Box>
+    )
+  }
+}
+
+export const YoastSnippet = styled(Box)``
+
+export const YoastContainer = styled(Flex)`
+  background-color: ${color("black5")};
+  padding: 0 20px;
+  height: ${space(4)}px;
+  border-bottom: 1px solid ${color("black10")};
+  font-size: 15px;
```

Comment:
why are we hardcoding a font-size here

Reply:
I think he was wondering if you actually needed it (the hardcoded value) because in the `<YoastContainer/>` styled component you have children components that use the `<Serif/>` component and pass `size=3` which I believe applies font-size.